e2e/cypress/integration/onboarding/use_team_invite_link_to_sign_up_spec.js

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @onboarding
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+import {getEmailUrl, getEmailMessageSeparator, reUrl, getRandomId} from '../../utils';
+
+describe('Onboarding', () => {
+    let testTeam;
+    const randomId = getRandomId();
+    const username = `user${randomId}`;
+    const email = `user${randomId}@sample.mattermost.com`;
+    const password = 'passwd';
+
+    const baseUrl = Cypress.config('baseUrl');
+    const mailUrl = getEmailUrl(baseUrl);
+
+    before(() => {
+        // # Do email test if setup properly
+        cy.apiEmailTest();
+
+        // # Update config to require email verification
+        cy.apiUpdateConfig({
+            EmailSettings: {
+                RequireEmailVerification: true,
+            },
+        });
+
+        cy.apiInitSetup().then(({team}) => {
+            testTeam = team;
+            cy.visit(`/${team.name}`);
+        });
+    });
+
+    it('MM-T398 Use team invite link to sign up using email and password', () => {
+        // # Open the 'Invite People' full screen modal and get the invite url
+        cy.get('.sidebar-header-dropdown__icon').click();
+        cy.get('#invitePeople').find('button').eq(0).click();
+        cy.get('.share-link-input').invoke('val').then((val) => {
```

Comment:
This will fail for server/local instance with license. Please add UI flow as commented at https://github.com/mattermost/mattermost-webapp/pull/6222#discussion_r475069039.